### PR TITLE
Throw on missing file in ChainEvent, revive unit tests that would have caught this

### PR DESCRIPTION
**Pull request by dan131riley**
#### PR description:

As recently revealed when an unit test input file disappeared, `fwlite::ChainEvent()` doesn't check the value of `TFile::Open()` and so segfaults on missing files.  This PR corrects the bug and revives and modernizes an old unit test that would have caught this.

#### PR validation:

Revived the unit test.  Verified that it segfaulted.  Modified `fwlite::ChainEvent()` to throw instead of faulting, verified that the unit test succeeded.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
Comparison differences are not related to changes in this PR.

File: DataFormats/FWLite/test/test.cppunit.cpp [hunk 1]
```diff
@@ -339,51 +339,36 @@ void testRefInROOT::testTwoGoodFiles() {
 }
 
 void testRefInROOT::testGoodChain() {
-  /*
-  TChain eventChain(edm::poolNames::eventTreeName());
-  eventChain.Add((tmpdir + "goodDataFormatsFWLite.root").c_str());
-  eventChain.Add((tmpdir + "good2DataFormatsFWLite.root").c_str());
+  std::vector<std::string> files{(tmpdir + "goodDataFormatsFWLite.root").c_str(),
+                                 (tmpdir + "good2DataFormatsFWLite.root").c_str()};
```
Comment (makortel) [inline DataFormats/FWLite/test/test.cppunit.cpp:343]: Would
```suggestion
  std::vector<std::string> files{tmpdir + "goodDataFormatsFWLite.root",
                                 tmpdir + "good2DataFormatsFWLite.root"};
```
work?

File: DataFormats/FWLite/test/test.cppunit.cpp [hunk 2]
```diff
@@ -339,51 +339,36 @@ void testRefInROOT::testTwoGoodFiles() {
...
-  for( int ev=0; ev<nev; ++ev) {
-    std::cout <<"event #" <<ev<<std::endl;    
-    eventChain.GetEntry(ev);
-    CPPUNIT_ASSERT(pOthers != nullptr);
-    CPPUNIT_ASSERT(pThings != nullptr);
-    checkMatch(pOthers->product(),pThings->product());
+
+void testRefInROOT::failChainWithMissingFile() {
+  std::vector<std::string> files{(tmpdir + "goodDataFormatsFWLite.root").c_str(),
+                                 (tmpdir + "2ndFileDoesNotExist.root").c_str()};
```
Comment (makortel) [inline DataFormats/FWLite/test/test.cppunit.cpp:359]: Same?
```suggestion
  std::vector<std::string> files{tmpdir + "goodDataFormatsFWLite.root",
                                 tmpdir + "2ndFileDoesNotExist.root"};
```

File: DataFormats/FWLite/test/test.cppunit.cpp (hunk 1 above)
Comment (dan131riley) [reply DataFormats/FWLite/test/test.cppunit.cpp]: Cut & paste from elsewhere in the test, fixed and fixed the other instances too

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**